server/lib/logger.ts

```diff
@@ -138,30 +138,66 @@ const yyyymmdd = () => {
   }${d} ${hour}:${minutes}:${seconds}`;
 };
 
+/**
+ * This function filter some known interfaces to avoid log hug objects
+ * @param data string | object
+ * @returns the data parsed 
+ */
+const parseData = (data: any) => {
+  let parsedData = data instanceof Error? { 
+    message: data.message,
+    stack: data.stack
+  } : data;
+
+  // when error is AxiosError, it extends from Error
+  if (data.isAxiosError) {
+    const { config } = data;
+    parsedData = {
+      ...parsedData,
+      config: {
+        url: config.url,
+        method: config.method,
+        data: config.data,
+        params: config.data,
+      }
+    };
+  }
+
+  if (typeof parsedData === 'object')
+    parsedData.toString = () => JSON.stringify(parsedData);
+
+  return parsedData;
+}
+
 /**
  * Main function to add a new log
  * @param {*} location File where the log is being thrown
  * @param {*} message Message to show
  * @param {*} level Optional, default is 'error'
  */
-export function log(location, message, level) {
+export function log(location, data, level) {
+  const parsedData = parseData(data);  
   initDirectory()
     .then(() => {
       if (allowed) {
         checkFiles();
-        wazuhlogger.log({
+        const options: any = {
           date: new Date(),
           level: level || 'error',
           location: location || 'Unknown origin',
-          message: message || 'An error occurred'
+          data: parsedData
```

Comment:
**question**: If we remove the `message` key, can this affects the frontend when displaying the Wazuh app logs in `Settings > Logs`?

Reply:
that's why the `getMessge` method was added in `settings-logs/logs.js#L79`